### ISSUE TITLE
Dataset API and Configuration

### ISSUE THREAD

**Issue by mmcdermott** [Oufattole/meds-torch#55]
I think that right now the dataset class has a lot of unnecessary complexity, so I suggest the following changes:

1. Move the text tokenization out of the dataset. Instead, text tokenization should happen separately, so that you only need to deal with integer ids and continuous values.
2. Simplify the input / output specification to be more functional, and less model-oriented. E.g., instead of having different collate types per model, have a more functional language that specifies what you want to do to the data.
3. We should have a simpler, more functional implementation of task specific functions to minimize their apparent complexity add.

Concretely, I envision the following dataset API:
1. Initialize, where you (potentially via the task label dataframe) built the appropriate index to define the getitem.
2. on each getitem, you grab the subject's data, slice to the appropriate time range, flatten the data as desired, then slice to the target max seq length. If prepending static data is desired, that should happen here. 
4. on collate, you merely vstack the ragged tensors and densify them. 

What are the options of this class? 
1. In what dimensionality do you want your output data? 2D, 3D, 4D (though these should be named based on the shapes expected, not numerical)?
2. Do you want numeric values or not? For EIC you don't need them, for example.
3. Do you want timedeltas or not?
5. Do you want static data pre-pended or included separately? This may be relevant as depending on the dimensionality this could change the sequence length, so it needs to happen here.
6. What style of subsequence sampling do you want?
7. What is your max and min sequence length constraints?
8. What task schema do you need to obey?
9. (Eventually) Do you want padded outputs or nested offset lists?

**Comment by mmcdermott** [comment]
This function is not generally needed and can be deleted I expect: https://github.com/Oufattole/meds-torch/blob/main/src/meds_torch/data/components/pytorch_dataset.py#L699

**Comment by mmcdermott** [comment]
This can be removed: https://github.com/Oufattole/meds-torch/blob/main/src/meds_torch/data/components/pytorch_dataset.py#L660

Instead, it can be done just within pytorch lightning or other things where you just stop the dataloader after a given number of batches (or actually even just setting the length manually). With the set stats gone it won't add any bias.

**Comment by mmcdermott** [comment]
Delete this: https://github.com/Oufattole/meds-torch/blob/main/src/meds_torch/data/components/pytorch_dataset.py#L588
Label schema should cover this and if not we should make it so

**Comment by mmcdermott** [comment]
Simplify this, but it needs to keep: https://github.com/Oufattole/meds-torch/blob/main/src/meds_torch/data/components/pytorch_dataset.py#L469

**Comment by mmcdermott** [comment]
delete this https://github.com/Oufattole/meds-torch/blob/main/src/meds_torch/data/components/pytorch_dataset.py#L33

**Comment by mmcdermott** [comment]
this can go https://github.com/Oufattole/meds-torch/blob/main/src/meds_torch/data/components/pytorch_dataset.py#L392 because it is all binary classification and in the label schema
